nowcasting_dataset/dataset/datasets.py

```diff
@@ -187,6 +206,79 @@ def __getitem__(self, batch_idx: int) -> example.Example:
 
         batch = example.to_numpy(batch)
 
+        if self.current_timestep_index is not None:
```

Comment:
I'm suprised this is necessary - or perhaps I'm not thinking right. The batch that is loaded should have the correct timesteps in it. 

If this option used incase the .nc has a large amount of data saved like -12 hours and + 12 hours, and then this can be used to reduce it to for example -3 hours, to + 1 hour

Becasue I love testing - I would be very tempted to move this into a seperate functions and right a unittest for it

Reply:
Yeah, this option is if the .nc has a large amount of data and we want to use only a subset of it, for example, some of the models that are larger might not fit in GPU memory if there is too much forecast or history timesteps, but we don't want to have to remake a second copy of the dataset just for that. I can split it into its own function and add tests though

Reply:
I made a unit test for `subselect_data` now